app/controllers/answers_controller.rb

```diff
@@ -2,16 +2,15 @@ class AnswersController < ApplicationController
 
   def new
     @lesson = Lesson.find(params[:lesson_id])
-    @lessons = Lesson.where(user_id: @lesson.user_id, category_id: @lesson.category_id )
+    #@lessons = Lesson.where(user_id: @lesson.user_id, category_id: @lesson.category_id )
     answer = @lesson.next_word
     if answer.nil?
       @correctcount = @lesson.choices.where(correct: true).count
-      @lessons.update(results: @correctcount)
-      #@results_nil_lesson = Lesson.where(user_id: @lesson.user_id, category_id: @lesson.category_id, results: nil )
-      #if @results_nil_lesson.count != 0
-        #@results_nil_lesson.destroy_all
-      #else
-      #end
+      @lesson.update(results: @correctcount)
+      @results_nil_lesson = Lesson.where(user_id: @lesson.user_id, category_id: @lesson.category_id, results: nil )
```

Comment:
@hironaooo 
You can use this statement as well:
`current_user.lessons.find_by(category_id: @lesson.category_id, results: nil) `